fsspec/asyn.py

```diff
@@ -16,7 +16,35 @@
 from .utils import is_exception, other_paths
 
 private = re.compile("_[^_]")
+iothread = [None]  # dedicated fsspec IO thread
+loop = [None]  # global event loop for any non-async instance
+_lock = None  # global lock placeholder
+
+
+def get_lock():
+    """Allocate or return a threading lock.
+
+    The lock is allocatted on first use to allow setting one lock per forked process.
+    """
+    global _lock
+    if not _lock:
+        _lock = threading.Lock()
+    return _lock
+
 
+def reset_lock():
+    """Reset the global lock.
+
+    This should be called only on the init of a forked process to reset the lock to
+    None, enabling the new forked process to get a new lock.
+    """
+    global iothread
```

Comment:
nitpick: the lists do not need to be `global`, since we are mutating them rather than assigning

Reply:
https://docs.python.org/3/reference/simple_stmts.html#the-global-statement
> It would be impossible to assign to a global variable without `global`

I've always understood this as "if you don't have the global statement, the change to the variable won't stick beyond the current function"

Reply:
https://stackoverflow.com/questions/7060711/accessing-module-level-variables-from-within-a-function-in-the-module

Reply:
```
In [1]: mylist = [None]

In [2]: def set_val(x):
   ...:     mylist[0] = x

In [3]: def get_val():
   ...:     return mylist[0]

In [4]: set_val(1)

In [5]: get_val()
Out[5]: 1
```